kotlin/codegen/src/test/java/com/squareup/moshi/kotlin/codegen/ksp/JsonClassSymbolProcessorTest.kt

```diff
@@ -0,0 +1,824 @@
+        }
+        """
+      )
+    )
+    assertThat(result.exitCode).isEqualTo(KotlinCompilation.ExitCode.COMPILATION_ERROR)
+    assertThat(result.messages)
+      .contains("supertype com.squareup.moshi.kotlin.codegen.JavaSuperclass is not a Kotlin type")
+  }
+
+  @Test
+  fun nonFieldApplicableQualifier() {
+    val result = compile(
+      kotlin(
+        "source.kt",
+        """
+          package test
+          import com.squareup.moshi.JsonClass
+          import com.squareup.moshi.JsonQualifier
+          import kotlin.annotation.AnnotationRetention.RUNTIME
+          import kotlin.annotation.AnnotationTarget.PROPERTY
+          import kotlin.annotation.Retention
+          import kotlin.annotation.Target
+
+          @Retention(RUNTIME)
+          @Target(PROPERTY)
+          @JsonQualifier
+          annotation class UpperCase
+
+          @JsonClass(generateAdapter = true)
+          class ClassWithQualifier(@UpperCase val a: Int)
+          """
+      )
+    )
+    assertThat(result.exitCode).isEqualTo(KotlinCompilation.ExitCode.COMPILATION_ERROR)
+    assertThat(result.messages).contains("JsonQualifier @UpperCase must support FIELD target")
+  }
+
+  @Test
+  fun nonRuntimeQualifier() {
+    val result = compile(
+      kotlin(
+        "source.kt",
+        """
+          package test
+          import com.squareup.moshi.JsonClass
+          import com.squareup.moshi.JsonQualifier
+          import kotlin.annotation.AnnotationRetention.BINARY
+          import kotlin.annotation.AnnotationTarget.FIELD
+          import kotlin.annotation.AnnotationTarget.PROPERTY
+          import kotlin.annotation.Retention
+          import kotlin.annotation.Target
+
+          @Retention(BINARY)
+          @Target(PROPERTY, FIELD)
+          @JsonQualifier
+          annotation class UpperCase
+
+          @JsonClass(generateAdapter = true)
+          class ClassWithQualifier(@UpperCase val a: Int)
+          """
+      )
+    )
+    assertThat(result.exitCode).isEqualTo(KotlinCompilation.ExitCode.COMPILATION_ERROR)
+    assertThat(result.messages).contains("JsonQualifier @UpperCase must have RUNTIME retention")
+  }
+
+  @Test
+  fun `TypeAliases with the same backing type should share the same adapter`() {
+    val result = compile(
+      kotlin(
+        "source.kt",
+        """
+          package test
+          import com.squareup.moshi.JsonClass
+
+          typealias FirstName = String
+          typealias LastName = String
+
+          @JsonClass(generateAdapter = true)
+          data class Person(val firstName: FirstName, val lastName: LastName, val hairColor: String)
+          """
+      )
+    )
+    assertThat(result.exitCode).isEqualTo(KotlinCompilation.ExitCode.OK)
+
+    // We're checking here that we only generate one `stringAdapter` that's used for both the
+    // regular string properties as well as the the aliased ones.
+    // TODO loading compiled classes from results not supported in KSP yet
```

Comment:
Replace this assertion with confirming how many files were generated?

Reply:
number of files wouldn't verify anything. This is about the number of adapter properties _in_ a file